meson.build

```diff
@@ -679,6 +679,10 @@ if time_epoch == -1
         source_date_epoch = run_command('sh', ['-c', 'echo "$SOURCE_DATE_EPOCH"']).stdout().strip()
         if source_date_epoch != ''
                 time_epoch = source_date_epoch.to_int()
+        elif run_command('[', '-d', '.git', ']').returncode() == 0
+                # If we're in a git repository, use the creation time of the latest git release tag.
+                latest_tag = 'v' + (meson.project_version().to_int() - 1).to_string()
```

Comment:
this is slightly sloppy, since if we are on the commit of a fresh release, i.e. the commit the newest tag points to, then we'll still use the version of the previous release.

Maybe use `git describe  --abbrev=0`.

Reply:
(There's also the problem that downstream distros sometimes append a suffix to the version, i.e. "245.7" or so, which would break the version maths you are doing here...)